src/commonMain/kotlin/exchange.dydx.abacus/processor/wallet/account/AccountProcessor.kt

```diff
@@ -952,6 +957,56 @@ internal class V4AccountProcessor(parser: ParserProtocol) : BaseProcessor(parser
         return modified
     }
 
+    internal fun receivedUnbonding(
+        existing: Map<String, Any>?,
+        payload: List<Any>?,
+    ): Map<String, Any>? {
+        if (payload == null) {
+            return existing
+        }
+        val modified = existing?.mutable() ?: mutableMapOf()
+        val unbondingDelegations: IMutableList<UnbondingDelegation> = iMutableListOf()
+
+        for (validator in payload) {
+            val validatorAddress = parser.asString(parser.value(validator, "validatorAddress")) ?: continue
+            val entries = parser.asList(parser.value(validator, "entries")) ?: continue
+
+            for (entry in entries) {
+                val completionTime = parser.asString(parser.value(entry, "completionTime")) ?: continue
+                val balance = parser.asString(parser.value(entry, "balance")) ?: continue
+
+                unbondingDelegations.add(UnbondingDelegation(validatorAddress, completionTime, balance))
+            }
+        }
+        modified.safeSet("unbondingDelegation", unbondingDelegations)
+        return modified
+    }
+
+    internal fun receivedStakingRewards(
+        existing: Map<String, Any>?,
+        payload: Map<String, Any>?,
+    ): Map<String, Any>? {
+        val rewards = parser.asList(parser.value(payload, "rewards"))
+        val total = parser.asList(parser.value(payload, "total"))
+        if (payload == null || rewards == null || total == null) {
+            return existing
+        }
```

Comment:
hmm do we need to be checking for `rewards == null || total == null` here? (Sorta feels like the for loops will just be skipped and we'll correctly set stakingRewards to null?

Reply:
Mainly did it due to typing issue, but I think its ok to return early if these are null